src/JustSaying/Messaging/MessageSerialization/SystemTextJsonSerializer`1.cs

```diff
@@ -0,0 +1,119 @@
+#if NET8_0_OR_GREATER
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using JustSaying.Extensions;
+using JustSaying.Messaging.MessageHandling;
+using JustSaying.Models;
+
+namespace JustSaying.Messaging.MessageSerialization;
+
+/// <summary>
+/// A class representing an implementation of <see cref="IMessageSerializer"/> for the <c>System.Text.Json</c> serializer.
+/// </summary>
+public class SystemTextJsonSerializer<T> : IMessageSerializer
+    where T : Message
+{
+    private readonly JsonSerializerOptions _options;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SystemTextJsonSerializer"/> class.
+    /// </summary>
+    public SystemTextJsonSerializer()
+        : this(null)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SystemTextJsonSerializer"/> class.
+    /// </summary>
+    /// <param name="options">The optional <see cref="JsonSerializerOptions"/> to use.</param>
+    public SystemTextJsonSerializer(JsonSerializerOptions options)
+    {
+        if (options == null)
+        {
+            options = new JsonSerializerOptions
+            {
+#if NET8_0_OR_GREATER
+                DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+#else
+                IgnoreNullValues = true,
+#endif
+            };
```

Comment:
Could cache this into a private static as the default, rather than create a new one per instance.